docs/dev/compiler-toolchain/solidity.md

```diff
@@ -0,0 +1,234 @@
+=======
+interface and I/O (input/output) methods. However, knowledge of these can be crucial for integration, debugging, or contribution purposes.
+>>>>>>> 9f8a9e6768f009dbbb4c254cdccc2e18312a7215
+
+The CLI supports several I/O modes:
+
+1. Standard JSON.
+2. Combined JSON.
+3. Free-form output.
+
+<<<<<<< HEAD
+All three modes use the standard JSON `solc` interface internally. It helps to reduce the complexity of the `zksolc`
+=======
+All three modes use the standard JSON `solc` interface internally which helps to reduce the complexity of the `zksolc`
+>>>>>>> 9f8a9e6768f009dbbb4c254cdccc2e18312a7215
+interface and facilitate testing.
+
+#### Standard JSON
+
+The `zksolc` standard JSON I/O workflow closely follows that of the official `solc` compiler. However, `zksolc` does not
+<<<<<<< HEAD
+support a significant part of it, as it only makes sense for compiling for EVM, and has no representation in the zkEVM
+=======
+support a significant part of the JSON workflow, specifically EVM configuration settings which have no representation in the zkEVM
+>>>>>>> 9f8a9e6768f009dbbb4c254cdccc2e18312a7215
+architecture.
+
+Additional zkEVM data is supported by `zksolc`, but omitted when passed to `solc`:
+
+- `settings/optimizer/enabled`: enables the LLVM optimizer. Disabling only makes sense for prototyping and debugging.
+- `settings/optimizer/mode`: sets the optimization mode. Available values: `0`, `1`, `2`, `3`, `s`, `z`. Default
+setting is `3`. See [LLVM optimizer](./llvm.md#optimizer).
+
+Unsupported sections of the input JSON, ignored by `zksolc`:
+
+- `sources/<file>/urls`
+- `sources/destructible`
+- `settings/stopAfter`
+- `settings/remappings`
+- `settings/optimizer/*`: the `solc` optimizer is always enabled and is not configurable.
+- `settings/evmVersion`
+- `settings/debug`
+- `settings/metadata`: we request metadata with default settings for consistency.
+- `settings/outputSelection/def`
+- `settings/modelChecker`
+
+Additional zkEVM data inserted by `zksolc`:
+
+<<<<<<< HEAD
+- `long_version`: the full `solc` version output
+- `zk_version`: the version of the `zksolc` itself
+- `contract/hash`: the hash of the zkEVM bytecode
+- `contract/factory_dependencies`: bytecode hashes of contracts `CREATE`d in the current one. [More detail here](../building-on-zksync/contracts/contract-deployment.md#note-on-factory-deps)
+=======
+- `long_version`: the full `solc` version output.
+- `zk_version`: the `zksolc` version.
+- `contract/hash`: the hash of the zkEVM bytecode.
+- `contract/factory_dependencies`: bytecode hashes of contracts created in the current contract with `CREATE`. [More details here](../building-on-zksync/contracts/contract-deployment.md#note-on-factory-deps).
+>>>>>>> 9f8a9e6768f009dbbb4c254cdccc2e18312a7215
+
+Unsupported sections of the output JSON, ignored by `zksolc`:
+
+- `contracts/<file>/<contract>/evm/bytecode`: replaced by the zkEVM object with only relevant data.
```

Comment:
“object” here is ambiguous; does it mean “object code” or “thing”?

Reply:
A JSON object.

Reply:
Changing to `replaced with a JSON object with zkEVM build data`.